OpenSim/Common/Component.cpp

```diff
@@ -178,14 +178,33 @@ void Component::connect(Component &root)
         AbstractConnector& connector = upd_connectors(ix);
         connector.disconnect();
         try{
-            connector.findAndConnect(root);
+            const std::string& compName = connector.get_connectee_name();
+            std::string::size_type front = compName.find("/");
+            if (front != 0) { // local name
+                // Local is considered (1) children 
+                const Component* comp = findComponent(compName);
+                if (!comp) {// and (2) siblings(same level)
```

Comment:
I don't understand these comments ("Local is considered children and siblings"). Perhaps clarify for posterity? :camel: